backend/controllers/usersController.go

```diff
@@ -1,211 +1,228 @@
-	object.Interests = body.Interests
-	object.Verified = body.Verified
-
-	// Update the object
-	result = db.Save(&object)
-	if result.Error != nil {
-		c.JSON(http.StatusBadRequest, gin.H{
-			"error": "Could not update object",
-		})
-
-		return
-	}
-
-	// Respond
-	c.JSON(http.StatusOK, object)
-}
-
-func (controller UsersController) Delete(c *gin.Context) {
-	db := database.GetDatabase()
-
-	// Get the existing object
-	id := c.Param("id")
-	var object models.Users
-	result := db.Find(&object, id)
-
-	if result.Error != nil {
-		c.JSON(http.StatusBadRequest, gin.H{
-			"error": "Could not retrieve object",
-		})
-
-		return
-	}
-
-	// Delete the object
-	result = db.Delete(&object)
-	if result.Error != nil {
-		c.JSON(http.StatusBadRequest, gin.H{
-			"error": "Could not delete object",
-		})
-
-		return
-	}
-
-	// Respond
-	c.JSON(http.StatusOK, gin.H{
-		"message": "Object deleted successfully",
-	})
-
-}
+package controllers
+
+import (
+	"net/http"
+
+	"github.com/VolunteerOne/volunteer-one-app/backend/database"
+	"github.com/VolunteerOne/volunteer-one-app/backend/models"
+	"github.com/VolunteerOne/volunteer-one-app/backend/service"
+	"github.com/gin-gonic/gin"
+)
+
+type UsersController interface {
+	Create(c *gin.Context)
+	All(c *gin.Context)
+	One(c *gin.Context)
+	Update(c *gin.Context)
+	Delete(c *gin.Context)
+}
+
+type usersController struct {
+	usersService service.UsersService
+}
+
+func NewUsersController(s service.UsersService) UsersController {
+	return usersController{
+		usersService: s,
+	}
+}
+
+var usersModel = new(models.Users)
+
+// Create ...
+func (controller usersController) Create(c *gin.Context) {
+	var err error
+
+	db := database.GetDatabase()
+
+	// Declare a struct for the desired request body
+	var body struct {
+		Id       uint
+		Handle   string
+		Email    string
+		Password string
+		// birthdate datatypes.Date `gorm: "NOT NULL"`
+		Birthdate string
+		FirstName string
+		LastName  string
+		// profilePic mediumblob,
+		Interests string
+		Verified  uint
+	}
+
+	// Bind struct to context and check for error
+	err = c.Bind(&body)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"error": "Request body is invalid",
+		})
+
+		return
+	}
+
+	// Create the object in the database
+	object := models.Users{
+		Handle:    body.Handle,
+		Email:     body.Email,
+		Password:  body.Password,
+		Birthdate: body.Birthdate,
+		FirstName: body.FirstName,
+		LastName:  body.LastName,
+		// ProfilePic: body.profilePic,
+		Interests: body.Interests,
+		Verified:  body.Verified,
+	}
+
+	result := db.Create(&object)
+
+	if result.Error != nil {
+
+		c.JSON(http.StatusBadRequest, gin.H{
+			"error": "Creation failed",
+		})
+
+		return
+	}
+
+	// Respond
+	c.JSON(http.StatusOK, object)
+}
+
+func (controller usersController) All(c *gin.Context) {
+	// var err error
+	db := database.GetDatabase()
+	var objects []models.Users
+
+	// Get objects from database
+	result := db.Find(&objects)

```

Comment:
The controllers should not be interacting with the db directly it should be done through the services layer